### PR TITLE
Atmos Overhaul + Extras

### PR DESCRIPTION
**Pull request by Jakksergal**
After whole **year** hopefully it'll be worth the wait.

Changes
- Completely redoes the greater atmos area to allow for more usable space and atmos experimentation.
- Adds lockable maintenance dorms. They're in a state of disrepair, making them the perfect place for some rough fun or a place to claim as your own.
- Adds a maint bar near the maintenance dorms. It only serves milk, trust me.
- Adds a public shower at the end of the dorm hallway. There are 2 private 'divisions' to get sudsy in.
- Adds various maint rooms and reconfigures older ones. The more hidden ones cater to heavier kinks.
- Reshapes the stairwell by the bar to allow access to the Z2 hallway as well as make the construction area near the tram not kidney-shaped.
- The surface hangar has been redone. There's space for the expedition shuttle to land but this functionality has not been implemented yet.
- Expands the station hangar to allow the aux shuttle to land there rather in space, making engine repair much easier if and when that functionality returns.

Minor changes:
- Moves the robotics wardrobe into robotics. No more will you have to snake into maintenance for a set of clothes.
- Shifts the research servers over a tad. They're now accessible from the hallway.
- The R&D lab now has a second maintenance door to the south.
- R&D lab room expanded just a smidgen, the table's been slightly uncluttered.

Images.
image
image
image

### CODE REVIEW

**Comment by Jakksergal** [comment]
.. Why does this require discussion when the last two overhauls did not? 

Also, all of this has been tested actively by the CitadelRP server, they've had this for the last 3 months.

**Comment by Very-Soft** [comment]
Alerting people to major changes to make sure they're okay and encouraging discussion about them via the tags is a good idea.

File: maps/tether/tether_areas2.dm [hunk 1]
```diff
@@ -320,6 +326,26 @@
...
+/area/crew_quarters/showers
+	name = "\improper Unisex Showers"
+	icon_state = "recreation_area_restroom"
+
+/area/crew_quarters/sleep/maintDorm1
+	name = "\improper Construction Dorm 1"
+	icon_state = "Sleep"
+
+/area/crew_quarters/sleep/maintDorm2
+	name = "\improper Construction Dorm 1"
```
Comment (Heroman3003) [inline maps/tether/tether_areas2.dm]: All the construction dorm areas have same name?

**Comment by Novacat** [comment]
The long and short of it is that I was not a developer during the last two overhauls, and, one of the things I want to do is get the community more involved in development directly, as opposed to complaining about problems after the fact.

**Comment by Screemonster** [comment]
needs less trashpiles. We have too many already. They're getting to the point where people can't scene in there because of the fucking lootrushers popping their heads in every few minutes.

**Comment by Very-Soft** [comment]
Yeah that many trash piles is gonna make that maint area really high traffic tbh. What is that, 11 new ones in the atmos maint stuff alone?

**Comment by Heroman3003** [comment]
I am just gonna list the things that I think should be adjusted, but general layout and idea behind it is good and net positive. The points are listed in no particular order

- I see that stealth buff to APCs in atmos, starting with super-capacity cells rather than regular ones. That stuff should probably be mentioned in the main post, considering its hard to notice both ingame and in review. Unless it really is a stealth buff attempt.

- Not even one, but two additional boxes of NIFsoft disks. Every other department only gets one per department, and engineering has least need for additionals of them all, I'd say (maybe after mining). I think neither should really be here.

- Two thermal regulators in the storage. There was one before. And engineering already had second one in the main storage. Any other heavy equipment amount changes done are alright, but this specific one is very powerful tool and considering its strength and rarity of need for second one in the first place, giving engineering third is really overkill. In fact one of those would better be replaced with two additional regular heaters.

- Construction area across from cafe probably shouldn't have either pump, scrubber or air alarm. Its unpowered anyway, but I do think it was better with absolutely zero infrastructure in the first place.

- New little hallway to stairs up to z-3 lacks cameras on the first floor.

- Arrivals laundry lost cameras added in one of latest additions.

- I am unsure on how much having regulators go from filters to storage will work out in terms of effectiveness compared to pumps, because while quicker on lower loads, heavy loads may end up getting clogged more easily potentially.

- I believe there is one missed regulator in O2/N2 to air mixer. All filters have them and there is no reason for mixer not to have one.

- I think evacuation route doors being red is rather misleading, red actually being much more associated with security and external airlocks. They were better as regular glass and maintenance doors.

- Surface 2 is now missing lightsout landmark.

And these ones is, I feel, more personal and much more subjective than rest of the points, but I will put it there anyway:

- While I do find it fun what you did with maint space, I do still think that four lockable protected dorms in maint is a bit of an overkill. I do like the idea, but I think making only one/two lockable and rest simply much more decrepit and rundown/unfinished and not lockable would make for a better layout in general. 

- Removing outer walls from the intake is also a move I do not really see as profitable. They worked relatively well at keeping wildlife away.

**Comment by Jakksergal** [comment]
@Screemonster @Very-Soft 
There were 7 trash piles in the previous configuration, this version has 13. 2x the trash piles in 3x the maintenance space, 4x if you count the atrium construction area. If anything we've lost overall trash pile density.

There aren't any trash piles in areas intended for scening? I'm not sure where you're seeing that.

@Heroman3003 
1. I did not buff the Atmos APC. Atmos, because of its huge power draw, has always had a super cell. The old atmos had a super cell but Aronai did not use the correct APC path for super cells and instead edited the variable of a normal APC.
2. There's only one additional NIF disk box. Atmos gets one, engineering gets one. 
3. The two thermal regulators was a product of this being in CitadelRP's hands for about 3 months. Will be fixed in next commit.
4. The construction area was given a pump/scrubber/air alarm because the alternative was using extremely wonky and often buggy endcaps that do not have a sprite in the editor. They aren't visually appealing either, as their sprite's misaligned on scrubber and distro pipes.
5. Will be fixed in next commit.
6. Will be fixed in next commit. What path did it have since I think that was your PR.
7. Regulators are not effective until the filters reach the pressure of the storage tanks *however* they perform much, much better at higher loads and without the power draw. The increase in draw from the filters pushing to a pressurized section of pipe is negated by the passive regulator's lack of power use but with the added bonus of allowing 200L/s rather than the 50L/s of a high power pump.

I chose this route because it's better at performing in the default atmos configuration which would previously really struggle with high volumes of gas, taking the entire shift to pump all the nitrogen in a siphoned fire into storage. Without setup, atmos struggles, but once the settings on all the roundstart active pumps are adjusted to their proper settings, atmos will be able to handle higher loads but still have room for improvement that an atmos tech can build for.
8. You're right, I see it. I had to convert the citadelRP room-based storage back into object-base storage, this was a carry-over I believe.
9. External airlock doors were used because they give the visual impression of showing that you're not supposed to be there while also attracting your eye while keeping the same visibility of a glass airlock. I had an issue with the normal glass airlocks because it looked like that was an area you were supposed to go into, like it was just a dirty hallway or damp room. I didn't quite feel like making a variant of the maintenance door specifically for evac.
10. Will be fixed in next commit.

1. There is no harm in allowing more space specifically for ERP. On CitadelRP where they average the same number of people that we encounter on highpop mid-week rounds, half of the dorm rooms were usually used with one room always being occupied for the remainder of the shift. These rooms double as personal projects, which further increases their use. Four is just enough to keep one room available most of the time.
2. There's no mechanical point to that.

**Comment by Heroman3003** [comment]
2. There ARE two new ones. One in locker room and one in breakroom. And I still think atmos shouldn't get more just because atmos, engineering is one department in terms of personnel and one box in engineering is sufficent still. Medical does not get second one in surface triage, security does not get second one on checkpoint, mining does not get second one in cargo, and I struggle to see why engineering should be any different.

4. Construction areas actually need neither and just cut off pipes do not leak any gas and fucntion fine without either endcap or scrubber/pump, while fitting into construction aesthetic. Endcaps were removed from existing construction areas precisely because of that.

**Comment by Jakksergal** [comment]
Oh, also adds NanoUI maps in that commit.

**Comment by MrSign** [comment]
Please, god. No more trash piles. Mice are already everywhere, we don't need more places for them to be breeding in!

**Comment by SpadesNeil** [comment]
>.. Why does this require discussion when the last two overhauls did not?

If I recall, the prior ones had discussions regardless.

Though I'mma remove staff discussion because this is really up to the community if they like these changes. I see no balance issues.

The one issue I do see is that these new maint dorm areas may not have the same event protection as normal dorms. So you might have things happen like, pest spawning, etc. I kind of don't want to fix that, but something you ought to know.

**Comment by Screemonster** [comment]
I'm also a little concerned that after the discord discussion about the hangar where you were warned about steamrolling ahead with changes and literally ignoring other people talking about their own discussions about the hangar, including a planned department shuffle at some point to get some use out of the less-used parts of the tether, along with the rants about the reserved area of the map (honestly the fuel depot is something that's a long ways away if anyone ever does bother with it so frankly I wouldn't care so much if you'd _asked_ about using that map area rather than _dictating_ that's what you were gonna do) you've gone ahead with it anyway - and done so in the same PR as the atmos changes which are far less contentious.

I don't like this kind of "you can have the thing you want but only if you accept the controversial thing too" packaging.

**Comment by nerdass** [comment]
as a matter of basic principle i'm against this being merged with the hangar modifications in place, because drawing a line in the sand and doing something you were told not to do because that area of the map was off limits, and bundling it in with a bunch of other neat things is a pretty shit thing to do.

you can wrap as many flowers around a turd as you want, it's not gonna change the fact that it's gonna smell like shit and roses. I'd legitimately feel uncomfortable playing on station with these changes in place.

**Comment by Sebbe9123** [comment]
I believe a future goal is to move away from trash piles and personally I dislike them strongly myself and would prefer them to be nowhere near areas intended for sceneing... especially since as I understand one of the reasons for having that area was for more extreme scenes.

**Comment by Novacat** [comment]
I am one of those people who does both maintenance scenes and is an avid trash pile looter. I believe the long term goal should be to divide our maintenance into two parts. One part for scenes, which would have no trash piles, and nerfed trash loot (no tactical jumpsuits, black dufflebags, bags of holding, etc.) The other part would have the trash piles and normal maintenance loot.

Ideally, that would kind of funnel the trash looters and the maintenance sceners away from each other, making both parties happy.

**Comment by Jakksergal** [comment]
@nerdass @Screemonster 
I was told no one gets to reserve areas of the map by the same person who said this space was reserved over a year ago. All they left was an empty plating room with nothing interesting in it, baring it off for everyone until they lost interest in developing for Virgo. The area serves no purpose, has no function, and the person responsible for implementing their changes hasn't spoken on the development channel in a month. I don't think that sounds like the activity of a person motivated to act on their reservations. 

There is still plenty of space for others to work. Even if the final product of it all is somehow the size of that entire 19x9 plating room, there is still room. I've stated in the development channel that I wouldn't be opposed to swapping the explorer's rooms to the other side of the hangar to give anyone *even more* space for an entirely hypothetical addition to exploration.

On the subject of the additions themselves, though, the aux shuttle *has* to be in an enclosed environment if you don't want it venting dock arm two. There doesn't exist a shuttle controller that's able to open one door at one destination and a different one at any other destination. Expanding the expedition hangar as opposed to making an entirely new hangar just makes the most sense. The alternative is to have one door on the shuttle but it would be facing the opposite way on the surface, towards a wall.

**Comment by Heroman3003** [comment]
Admittedly, all the shuttle and exploration bay stuff seems pretty out of scope of the "atmos rework" part of PR. Just saying, those two do not have much in common and did not at all have to be packaged in same wrapping.

**Comment by Jakksergal** [comment]
@Heroman3003 You're not wrong. The original PR only included the surface hangar because, at the time, I didn't know about the shuttle's habbit of venting itself every time it went to the station. Rather than do some wonky copy-and-pasting of the old hangar situation it seemed simpler to fix the problem by continuing the trend of renovations. But that's apparently not the case.

**Comment by Screemonster** [comment]
The fact that you're making it a principled "NOBODY GETS TO RESERVE SPACE" comes across like you're putting it there specifically out of spite and bundling it with another change to try and force us to merge it.

I'd rather have a mediocre map than a mapper who thinks he can hold his changes hostage.

**Comment by Jakksergal** [comment]
The thing is *I'm not*. 

The only reason against the changes so far has been about how the space was reserved but no one's acted on it and the person responsible for those reservations has left Virgo's development. The others saying they've had plans to do something with it haven't produced anything yet so I'm not stepping on toes by introducing conflicts and I've not removed their ability to make those changes, whatever they might be. 

I don't see any objective statements being made about the content itself but rather the parameters surrounding it. How I operated on a hunch that the space is no longer being used rather than asking someone who is no longer around if I could have their permission to do something that went against their year-old claims. How I'm supposedly steamrolling other people's content by editing space they *also* shouldn't be touching. It really feels like there's a double standard in place.

All that being said I've found a less elegant solution to the whole controversy of sacred tiles and it's coming in the next commit.

**Comment by Sebbe9123** [comment]
I think an important difference here some of us are probably getting hung up on is the difference between a head dev or a dev at all saying "don't change this place, I have something planned down the pipeline and a random dude like me saying don't touch x I need it for something I have planned.
I'm pretty confident the refueling thing Aro wanted to do was in the frameworks etc he just wanted to finish the goopies first, that does not matter anymore but the way you are presenting stuff does seem slightly spiteful.

Anyways, trash piles bad, maintenence scenes good.

**Comment by Jakksergal** [comment]
I would have thought that it would be safe to say a former head dev's claims would stop being relevant when they stopped developing for the server but this whole debacle has shown me otherwise.

I wasn't trying to put the shuttle there *because* someone reserved it years ago. I was putting it there because it just made the most sense to expand a current hangar instead of make an entirely new one budding off the edge of the station like a tumor.

**Comment by Novacat** [comment]
Personally, I think rather than expanding the current hangar, move those changes to a general hangar overhaul. One of the things I would like to see done, is to move the AI upload/core where the Chapel is now, move the Chapel to where the Hangar is now, and move the Hangar to all that space where the AI Core and old IAA Office used to be.

**Comment by Jakksergal** [comment]
Yeah, that makes sense and I'd be down to do that in the future. It makes sense to bundle all those things together but I didn't see anything objectively wrong with including a module of that earlier than the rest. In this instance it would make the transition just a tad bit easier now that the shuttle is configured to not take the floor with it when it leaves from a hangar environment and actually docks properly. Whoever moves the shuttle bay now has to do some duplication of effort.

**Comment by Heroman3003** [comment]
Well, one of the NIFsoft boxes still remains, but whether its okay to increase something that has before given me impression of intentionally limited, will be up to decision of actual admins, just puttint it out there.

Also, still has not heard back on potentially still using the simple pipe cutoff without the pumps on the end, since the pumps will end up being depowered.

**Comment by Heroman3003** [comment]
Oh, and one last thing I only now realized I missed. New atmos starts with filled phoron canisters (before it only had empty ones). That is a balance concern, but a very minor one, IMO, so not to me to decide. And for QoL, it is missing the pre-made empty canisters that old atmos held in plentiful amount, which is rather sad to see.

**Comment by Jakksergal** [comment]
I personally don't think it matters that much if engineering gets more nifsoft disks than there are job slots. Not everyone has a NIF and those atmos techs that do shouldn't have to walk all the way up to engineering on roundstart, they've already got to do a lot at that time.

Same goes for the vents in the construction area. It being there doesn't cause any harm but if an engineer wants to make a project, particularly a novice engineer, they've already got atmos covered.

**Comment by Heroman3003** [comment]
The canisters canisters. Ones you wrench on ports, carry around, etc. Not the big storage tanks. Those are fine. But Previous atmos had 7 empty canisters. I do not think it needs that many, but its good to have some spares. And old atmos had no phoron canisters with actual gas within, only empty ones for filling uisng intake.

**Comment by Screemonster** [comment]
I've made a separate issue for the "what to do with the hangar" thing, see #4894

**Comment by SpadesNeil** [comment]
I don't give a shit about who reserved an area for what. Either get it done or your shit gets overwritten by someone else. ¯\_(ツ)_/¯

So can someone summarize the issues besides that?

**Comment by Heroman3003** [comment]
Well, the whole reserved area space got brought back in way it was before so that is not a problem anymore.

As for the things that were brought up, mostly my own, with different levels of subjectivity are:

- Too many new trash piles

- Additional NIFsoft box that was not in old atmos

- Lack of premade empty canisters

- Canisters with phoron start with actual gas in them, unlike before where atmos had no roundstart phoron (I believe it was perfectly intentionally before)

- Still no regulator/pump after the air mix mixer into the storage tanks, inconsistant with before and with existing new filters.

- Construction area across cafe has useless pump/scrubber/air alarm that will not work since area is unpowered instead of pipes cutting off like most other construction areas

- Colors of two 'evacuation route' airlocks look a bit out of place

Most of those, once again, are stuff I brought up and pretty subjective. I believe biggest (and most objective) things here are lack of canisters to fill, lack of regulator after the mixing omni and overabundance of trash piles. Rest are matter of opinions.

**Comment by Jakksergal** [comment]
@Heroman3003 
1. Added trash piles was reduced to +3 from the previous version. For the added main space I think that's fair. 
2. It's pretty subjective to have an additional NIFsoft box. Not everyone has a NIF and those that do probably don't want to go up to engineering for one. I don't have actual use rates but I hardly see disks missing from them when I go to grab mine.
3. Canisters can be made with metal, that's why all original atmosias had a stack of metal in them.
4. I'm guessing there were no phoron canisters in the old atmos because Aronai did not make room for an enclosed space to keep them. I'm fairly certain most atmos' have them. 
5. Will fix.
6. Bundling this up with the mixer fixer.
7. That's the point. They're meant to show you the way. It's a pretty crucial emergency route and I didn't it to look like a dirty hallway with the previous glass doorway as I've already said.

**Comment by SpadesNeil** [comment]
All seems fine to me.

**Pull request merged**